src/main/java/com/minecolonies/util/Log.java

```diff
@@ -12,16 +10,5 @@
     /**
      * Mod logger
      */
-    public static final Logger logger = LogManager.getLogger(Constants.MOD_ID);
-
-    /**
-     * Generates a logger for a specific class
-     *
-     * @param clazz Class to generate logger for
-     * @return Created {@link Logger}
-     */
-    public static Logger generateLoggerForClass(Class<?> clazz)
-    {
-        return LogManager.getLogger(Constants.MOD_ID + "::" + clazz.getSimpleName());
-    }
+    public static Logger logger = null;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Make this "public static logger" field final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1444)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make logger a static final constant or non-public and provide accessors if needed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)